network/src/protocols/identity.rs

```diff
@@ -1,44 +1,14 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-//! Protocol used to identify key information about a remote
-//!
-//! Currently, the information shared as part of this protocol includes the peer identity and a
-//! list of protocols supported by the peer.
+//! Protocol used to exchange supported protocol information with a remote.
+
 use crate::protocols::wire::handshake::v1::HandshakeMsg;
 use bytes::BytesMut;
 use futures::io::{AsyncRead, AsyncWrite, AsyncWriteExt};
-use libra_types::PeerId;
 use netcore::framing::{read_u16frame, write_u16frame};
 use std::io;
 
-/// The PeerId exchange protocol.
-pub async fn exchange_peerid<T>(own_peer_id: &PeerId, socket: &mut T) -> io::Result<PeerId>
-where
-    T: AsyncRead + AsyncWrite + Unpin,
-{
-    // Send serialized PeerId to remote peer.
-    let msg = lcs::to_bytes(own_peer_id).map_err(|e| {
-        io::Error::new(
-            io::ErrorKind::InvalidData,
-            format!("Failed to serialize identity msg: {}", e),
-        )
-    })?;
-    write_u16frame(socket, &msg).await?;
-    socket.flush().await?;
-
-    // Read PeerId from remote peer.
-    let mut response = BytesMut::new();
-    read_u16frame(socket, &mut response).await?;
-    let remote_peer_id = lcs::from_bytes(&response).map_err(|e| {
-        io::Error::new(
-            io::ErrorKind::InvalidData,
-            format!("Failed to parse identity msg: {}", e),
-        )
-    })?;
-    Ok(remote_peer_id)
-}
-
 /// The Handshake exchange protocol.
 pub async fn exchange_handshake<T>(
```

Comment:
unrelated to this PR, but IIUC this is only to negotiate supported protocol, which is just a part of the handshake. So I find it weird to call that a handshake :D (and we already have a handshake in noise)

Reply:
agreed, handshake is not descriptive enough. it's a protocol+version negotiation handshake.

Reply:
I think we also need to figure out what this really means... might be worth having an adhoc sync next week...
but what does it mean to support a protocol? What if I can send queries on that protocol (i.e., a client) but have no service? Do I express this in the handshake or is the handshake only for server protocols?

My gut feeling is that we have a lot of flexibility here and this is more of how we internally handle this versus what the wire spec enforces.